typing/env.ml

```diff
@@ -703,20 +703,28 @@ let find_name_module ~mark name tbl =
 
 let add_persistent_structure id env =
   if not (Ident.persistent id) then invalid_arg "Env.add_persistent_structure";
-  if not (Current_unit_name.is_name_of id) then
-    let summary =
+  if not (Current_unit_name.is_name_of id) then begin
+    let material =
```

Comment:
Is there a comment somewhere that explains why this definition is doing? (If I remember the previous PR correclty, we are checking some shadowing condition. An explanation in the code would be helpful, is it somewhere else?)

Reply:
Addressed